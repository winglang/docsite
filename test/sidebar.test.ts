const { reverseSidebarItems } = require('../src/utils/sidebar');

const testItems = [{ "type": "doc", "id": "getting-started" }, {
  "type": "doc",
  "id": "learning-winglang"
}, {
  "type": "category",
  "label": "Reference",
  "items": [{ "type": "doc", "id": "reference/intro" }, { "type": "doc", "id": "reference/WingSDK" }, {
    "type": "doc",
    "id": "reference/WingConsole"
  }, {
    "type": "category",
    "label": "Previous Versions",
    "items": [{
      "type": "category",
      "label": "WingSDK",
      "items": [{
        "type": "doc",
        "id": "reference/previous_versions/WingSDK/0.0.39",
        "label": "0.0.39"
      }, { "type": "doc", "id": "reference/previous_versions/WingSDK/0.0.40", "label": "0.0.40" }, {
        "type": "doc",
        "id": "reference/previous_versions/WingSDK/1.1.39",
        "label": "1.1.39"
      }, { "type": "doc", "id": "reference/previous_versions/WingSDK/1.2.0", "label": "1.2.0" }],
      "link": { "type": "generated-index", "description": "Reference Documentation" }
    }],
    "link": { "type": "generated-index", "description": "Previous Versions" }
  }],
  "link": { "type": "generated-index", "description": "Reference Documentation" }
}, { "type": "doc", "id": "resources" }];

describe('sidebar', () => {
  it('reverses only the previous versions category and popping the top', () => {
    const results = reverseSidebarItems(testItems);
    expect(results).toMatchSnapshot()
  });
});
