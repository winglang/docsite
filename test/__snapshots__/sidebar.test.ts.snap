// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar reverses only the previous versions category 1`] = `
Array [
  Object {
    "id": "getting-started",
    "type": "doc",
  },
  Object {
    "id": "learning-winglang",
    "type": "doc",
  },
  Object {
    "items": Array [
      Object {
        "id": "reference/intro",
        "type": "doc",
      },
      Object {
        "id": "reference/WingSDK",
        "type": "doc",
      },
      Object {
        "id": "reference/WingConsole",
        "type": "doc",
      },
      Object {
        "items": Array [
          Object {
            "items": Array [
              Object {
                "id": "reference/previous_versions/WingSDK/1.2.0",
                "label": "1.2.0",
                "type": "doc",
              },
              Object {
                "id": "reference/previous_versions/WingSDK/1.1.39",
                "label": "1.1.39",
                "type": "doc",
              },
              Object {
                "id": "reference/previous_versions/WingSDK/0.0.40",
                "label": "0.0.40",
                "type": "doc",
              },
              Object {
                "id": "reference/previous_versions/WingSDK/0.0.39",
                "label": "0.0.39",
                "type": "doc",
              },
            ],
            "label": "WingSDK",
            "link": Object {
              "description": "Reference Documentation",
              "type": "generated-index",
            },
            "type": "category",
          },
        ],
        "label": "Previous Versions",
        "link": Object {
          "description": "Previous Versions",
          "type": "generated-index",
        },
        "type": "category",
      },
    ],
    "label": "Reference",
    "link": Object {
      "description": "Reference Documentation",
      "type": "generated-index",
    },
    "type": "category",
  },
  Object {
    "id": "resources",
    "type": "doc",
  },
]
`;
